### PR TITLE
AB#3852 AB#3853 -- updated child flow adult/child review pages

### PR DESCRIPTION
**Pull request by AdamP-HRSDC-RHDCC**
### Description
Reviewed and updated child flow pages for review-adult-information and review-child-information

### Related Azure Boards Work Items
[AB#3852](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3852)
[AB#3853](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3853)

### Screenshots (if applicable)
image
image

Dummy payloads for testing (Disable validation)

Adult:
```
const userInfo = {
      firstName: "test",
      lastName: "test",
      phoneNumber: "test",
      altPhoneNumber: "test",
      preferredLanguage: "test",
      birthday: "test",
      sin: "000000000",
      martialStatus: "test",
      email: "test",
      communicationPreference: "test",
    };
  const spouseInfo = state.partnerInformation
    ? {
        firstName: 'test',
        lastName: 'test',
        birthday: 'test',
        sin: '000000000',
        consent: 'test',
      }
    : undefined;

  const preferredLanguage = 'test';

  const mailingAddressInfo = {
    address: 'test',
    city: 'test',
    province: 'test',
    postalCode: 'test',
    country: 'test',
    apartment: 'test',
  };

  const homeAddressInfo = {
    address: 'test',
    city: 'test',
    province: 'test',
    postalCode: 'test',
    country: 'test',
    apartment: 'test',
  };
```

Children:
```
const children = [
    {
      id: 'string',
      dentalBenefits: {
        hasFederalBenefits: true,
        federalSocialProgram: 'string',
        hasProvincialTerritorialBenefits: true,
        provincialTerritorialSocialProgram: 'string',
        province: 'string',
      },
      dentalInsurance: true,
      information: {
        firstName: 'test',
        lastName: 'test',
        dateOfBirth: 'test',
        socialInsuranceNumber: '000000000',
        isParent: true,
      },
    },
    {
      id: 'string2',
      dentalBenefits: {
        hasFederalBenefits: true,
        federalSocialProgram: 'string',
        hasProvincialTerritorialBenefits: true,
        provincialTerritorialSocialProgram: 'string',
        province: 'string',
      },
      dentalInsurance: true,
      information: {
        firstName: 'test2',
        lastName: 'test2',
        dateOfBirth: 'test',
        socialInsuranceNumber: '000000000',
        isParent: true,
      },
    },
  ];
```

### CODE REVIEW

File: frontend/app/mappers/benefit-application-service-mappers.server.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export function toBenefitApplicationRequest({
           PrivateDentalInsuranceIndicator: dentalInsurance,
           DisabilityTaxCreditIndicator: disabilityTaxCredit,
           LivingIndependentlyIndicator: livingIndependently,
-          InsurancePlan: toInsurancePlan(dentalBenefits),
+          InsurancePlan: dentalBenefits ? toInsurancePlan(dentalBenefits) : undefined,
```
Comment (sebastien-comeau) [inline frontend/app/mappers/benefit-application-service-mappers.server.ts:52]: Uses Logical AND (&&) operator

```suggestion
          InsurancePlan: dentalBenefits && toInsurancePlan(dentalBenefits),
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/public/locales/fr/apply-child.json [hunk 2]
```diff
@@ -511,20 +511,38 @@
...
+    "dental-title": "{{child}} accès à l'assurance dentaire",
+    "dental-insurance-title": "Accès à une assurance ou une couverture dentaire",
+    "dental-insurance-change": "Modifier la réponse à «\u00a0accès à l'assurance dentaire\u00a0»",
+    "dental-benefit-title": "Accès à des prestations dentaires gouvernementales",
+    "dental-benefit-change": "Modifier la réponse à «\u00a0soins dentaires gouvernementaux\u00a0»",
+    "dental-benefit-has-access": "Accès à des prestations dentaires gouvernementales\u00a0:",
+    "dental-benefit-has-no-access": "Accès à des prestations dentaires gouvernementales\u00a0:",
+    "continue-button": "Continuer",
     "back-button": "Retour",
+    "exit-button": "Quitter la demande",
```
Comment (sebastien-comeau) [inline frontend/public/locales/fr/apply-child.json:545]: That key shouldn't exist in the "child" flow context for "review-child-information".

File: frontend/public/locales/en/apply-child.json [hunk 3]
```diff
@@ -512,20 +512,38 @@
...
+    "dental-title": "{{child}} access to dental insurance",
+    "dental-insurance-title": "Access to other dental insurance or coverage",
+    "dental-insurance-change": "Change answer to dental insurance",
+    "dental-benefit-title": "Access to government dental benefit",
+    "dental-benefit-change": "Change answer to public dental benefits",
+    "dental-benefit-has-access": "Has access to the following benefits:",
+    "dental-benefit-has-no-access": "Has access to the following benefits:",
+    "continue-button": "Continue",
     "back-button": "Back",
+    "exit-button": "Exit application",
```
Comment (sebastien-comeau) [inline frontend/public/locales/en/apply-child.json:546]: That key shouldn't exist in the "child" flow context for "review-child-information".

File: frontend/app/mappers/benefit-application-service-mappers.server.ts (hunk 1 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/mappers/benefit-application-service-mappers.server.ts]: Changed to &&

File: frontend/public/locales/fr/apply-child.json (hunk 2 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/public/locales/fr/apply-child.json]: Removed unused translation

File: frontend/public/locales/en/apply-child.json (hunk 3 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/public/locales/en/apply-child.json]: Removed unused translation

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**